### ISSUE TITLE
Wrath of Cenarius proc mechanics

### ISSUE THREAD

**Issue by Alienlex** [LightsHope/issues#13]
Greetings Light's Hope staff! I'd like to start off by saying I love how you are preserving the vanilla experience. Progressing through AQ 40 these last couple of weeks has been a blast.

So I farmed Cenarion Circle exalted reputation on my shadow priest for the Wrath of Cenarius ring. If functioning as intended it is supposed to work wonders with channeled spells like Mind Flay and Arcane Missiles. This is due to it having a chance to proc off of ticks of channeled spells, not just the initial cast. However after getting the ring I immediately noticed this wasn't working correctly with Mind Flay. So I decided to test with low rank smite to see if the procrate is correct and indeed it seems to be 5% as stated by most sources I've found. 

I had done quite a bit of research already because the item is time consuming to farm, and aware of random coding issues sometimes with things like proc rates and patches with individual items, I decided to dig deeper and figure out what the root of the problem might be if there is any problem at all or if its working as intended. 

So far the information I've gathered from sources provided seems to summarize into this-

Patch 1.91- release of item. reports of issues that it isnt working with some channeled spells dmg ticks or not working for some channeled spells at all(drain life and blizzard were both mentioned more than once). reports that it procs off of non damaging spells as well. 

Patch 1.10- fixed bug that caused it to proc from non damaging spells like Detect Magic, and fixed to work as intended for most spells, but became bugged with Arcane Missiles and Blizzard to not work at all.

Patch 1.12- fixed to work with all channeled spells as intended

As a shadow priest there is very little information available on whether or not Mind Flay is bugged with this item in patch in 1.91. Based on what I've read it worked in 1.10 and beyond so maybe this bug report is premature. From my own personal testing on Light's Hope it seems to proc off the initial cast of Mind Flay just fine, but not the channeled ticks. I just wanted to make the staff aware of this and provide as many details as possible. However you all decide to handle it is cool with me :). I have a slight feeling when ya'll patch 1.10 and 1.12 it might just apply the fix naturally. I've noticed some things fixed in this manner so it wouldn't suprise me. Much love to you all. The work you've done with Lights Hope is priceless

Sources-
Allakhazam 
https://web.archive.org/web/20071118171352/http://wow.allakhazam.com:80/item.html?witem=21190
Thottbot
https://web.archive.org/web/20060428035807/http://www.thottbot.com:80/?i=52574

--

**Comment by brotalnia** [comment]
Currently that proc is hack-fixed, somebody who understands the spell proc system better needs to do it properly.

https://github.com/ratkosrb/server/blob/development/src/game/UnitAuraProcHandler.cpp#L278

**Comment by GuybrushGit** [comment]
https://vanilla-twinhead.twinstar.cz/?issue=6587

### Periodics
- Current behaviour: Proc once on initial cast
- Correct behaviour: Most likely proc once on initial cast

>it can proc on any DoTs initial cast, but not on DoT ticks.
>http://wowdata.buffed.de/item/Ruestung/Verschiedenes/Zorn-des-Cenarius-21190#comments

>this includes DoT's(one from the debuff landing but not from the ticks)
>https://web.archive.org/web/20070313104450/http://wow.allakhazam.com/db/item.html?witem=21190


 ### Channeled Direct
- Current behaviour: Proc on each tick but not on initial cast
- Correct behaviour: Proc on each tick, unsure if initial cast should

>yes on individual ticks.. meaning ive had it proc on the third missile and so on.
>https://web.archive.org/web/20070906075803/http://thottbot.com/i21190


### AoE Direct
- Current behaviour: Procs once on initial cast, even if no targets are hit. Procs once again on spell hit, regardless of how many targets are hit.
- Correct behaviour: Shouldn't proc on initial cast. Most likely should proc once per target.

> it does proc often off aoes.. just about every other iae or blastwave of multiple targets.. 
>https://web.archive.org/web/20060428035807/http://www.thottbot.com:80/?i=52574


### AoE Channeled
- Current behaviour: Procs once on initial cast, even if no targets are hit. Procs again on every damage dealt.
- Correct behaviour: Shouldn't proc on initial cast. Unsure if it should proc on all damage dealt or only once per target.

> procs A TON when I use AE on the trash mobs  

>multiple proc chance while AoE'ing sevral targets  

>https://web.archive.org/web/20070313104450/http://wow.allakhazam.com/db/item.html?witem=21190

>-it rocks for aoe'ing... the more targets you aoe, the more proc you'll see. 
>https://web.archive.org/web/20060428035807/http://www.thottbot.com:80/?i=52574

**Comment by Alienlex** [comment]
Thank you very much for checking this out in such detail. To clarify the main issue im having here is the following:

Channeled Direct
Current(Lightbringer): Proc on initial cast but not on each tick
Correct: Proc on each tick, unsure if initial cast should

I don't have any means to test it with aoe as I am a shadow priest, but this working with channeled spells properly makes or breaks the item for me. I wish I was a programmer or had better knowledge of the subject so I could provide more info of some kind. If it helps at all as far as I know and have read this ring works the same mechanically in vanilla(once the bugs were fixed by Blizzard) as it did in TBC. Thanks again I hope a solution can be reached soon :)

**Comment by GuybrushGit** [comment]
The only Channeled Direct spell in the same is Arcane Missiles. Priest spells are periodics.

**Comment by Alienlex** [comment]
Ok this clears things up for me then. It should only proc off the initial cast of periodics of course, and if Mind Flay is classified as a periodic in vanilla then I was indeed misinformed. I have not tested it for mages obviously, and if it works as intended for missiles and aoe now then there's no need to make any changes here. The proc rate of 5% seems correct, and it correctly procs off initial casts of SWP, Mind Flay, Mind Blast, Smite etc just fine. Thank you for your time much love

**Comment by theloras** [comment]
@brotalnia - from what I remember when you worked on it initially you weren't able to get Paladin Seals to trigger the proc but were able to get both Judgement of Command and Righteousness to trigger as well as Consecration and Holy Shock on top of our undead DD spells such as Exorcism and Holy Wrath.

I also remember from the original LH github repository another report where it was stated that the fire/nature dmg on each swing items should trigger the proc as well (ie: Fiery Retributer, Blazefury Medallion, Fiery Plate Gauntlets, Storm Gauntlets) since their procs are individual instant spell casts on each swing.

**Comment by fidbugs** [comment]
Thanks for all the digging @GuybrushGit .

I would like to add regarding the AoE channeled mechanic - does it proc on every damage or just once per target?

Best lead is comment on WoWhead and wording of the tooltip itself.

https://gyazo.com/21aecc8dee633ddde93a8531fb901d8f

Clearly states "it will refresh itself in AoE situations" - which seems most plausible if proc check is made on every damage done, not only on initial one per target. In that case blizzard/consecration/etc. would have chance to trigger on every tick on every mob.

Also wording says "when your harmful spell _LAND_" - everything stated above in Guybrush's analasis leads us to conclusion that every landing damage from a spell is a event that has chance to trigger the proc. 

That being said, interesting thing are seals, enchants(Lifesteal, Fiery, Frost/Shadow oil, etc.), item procs(Maelstrom), things like Blazefury Medallion and food buffs(Dragonbreath Chili).

My two cents:

TLDR "if spell, if lands, then trigger proc"

**Seals** - since seals are treated as spells, it should proc when a seal hits target, not when it s cast(Righteousness, Command - note - should also proc on relevant Judgements). 

As a point in that direction, ZHC looses stacks when seals/judgements hit(land), thus it would make perfect sense they trigger ring since they are considered spells and they _LAND_. 

Reference to patch 1.10 notes http://wowwiki.wikia.com/wiki/Patch_1.10.0

**Enchants** - my best lead here is Vengance talent logic, if game treats enchants as a spell that hits target, it should be able to trigger ring proc. Also to note is the spell damage scaling that oils receive - it should strengthen the argument for "spell" treatment of at least oil temporary enchants. Tho it would make much more sense that Fiery, Lifesteal and Shadow/Frost oil have same mechanic.

**Item Procs** - similar to enchants, most of item procs are damage spell effects(Malstrom, Ashbringer, etc.) - in this case following same logic it should be event that riggers the ring.

**Blazefury and similar** - main point here is infamous video of Storm Gauntlets and grounding totem interaction. If game treats them as "casted spell" - they should also trigger ring to proc.

https://www.youtube.com/watch?v=tDi_g4XDwBU

**Dragonbreath chilli** - same logic as things above - it scalles with spell damage, thus spell, thus should trigger. Also since it has AoE potential should be treated same as AoE Dirrect.

So for conclusion, id just highlight things from paladin perspective. Should proc of seal of command/righteousness, judgements, wrath(AoE direct), exorcism, consecrate(AoE channeled).

**Comment by theloras** [comment]
"As a point in that direction, ZHC looses stacks when seals/judgements hit(land), thus it would make perfect sense they trigger ring since they are considered spells and they LAND."

no they don't - ZHC loses stack charges if you use JUDGEMENT - not on each individual Seal landing during the 20 sec period

**Comment by Alienlex** [comment]
Sorry for the premature closing. A couple of things I can add on further testing. It was proccing from wanding in my AQ20 run last night. The proc can occur twice in a row, so no internal cooldown to be found. 

In regards to Consecrate and Blizzard, they do not cause pushback and should be periodic damage spells much like Mind Flay. Correct me if I'm wrong but if ticks of damage from them should cause the ring to proc then it should work like that for Mind Flay as well. This is what I've always heard in the stories from others, read in comments about this ring. I understand theyre classified as periodic spells but based on the comments it's pretty clear that different spells were treated differently in regards to how they proc the ring in the patch notes from 1.91-1.12. Ive always heard it was amazing for shadow priests, mages, paladins and even warlocks due to the niche use with channeled casts. I understand it's hard to make any changes without 100% clarification but finding that might be impossible and all evidence seems to point in this direction.

**Comment by theloras** [comment]
One thing to note though is that Consecration i think is in a category unto itself as it's more like a fire and forget AoE over time spell rather than a channeled spell like Mind Flay or Blizzard.

**Comment by Alienlex** [comment]
Any word on a fix for this ring in the patch? Testing has confirmed that it procs from each individual damage tick of channeled spells for blizzard, arcane missiles, hellfire and rain of fire and is working properly for those spells. However Mind Flay, Drain Life, and Siphon Soul were clearly left out and only proc on initial cast. It would be really awesome if us shadow priests and locks could get some love here so our rings are actually useful. Much love to u guys!

**Comment by GuybrushGit** [comment]
Mind Flay, Drain Life, and Siphon Soul proccing only on initial cast is the correct behaviour.

**Comment by Alienlex** [comment]
Well then we'll have to agree to disagree. And I'll probably vendor my ring if it doesn't magically and randomly get fixed in the 1.10 update. I was encouraged to farm this ring on my shadow priest when vanilla was retail because of its mechanic with channeled spells, but never did because I quit before Naxx came out. I'm aware evidence from other private servers isn't counted as evidence, and I wish this ring and shadow priests together weren't so rare that I can't prove it with actual retail video evidence, but my friend played a shadow priest on Rebirth and this is how the ring worked. And on other servers as well, I found this thread - https://truewow.org/forum/viewtopic.php?t=34089

I probably should have done more research because just now I've found this thread where a user said the ring didn't work correctly on Nostalrius for shadow priests, so its no surprise to me that it works the same way here.-http://forum.nostalrius.org/viewtopic.php?f=39&t=36842

I understand it's a "we dont have enough evidence to change this." kind of like challenging a call in football or baseball and it's super unfortunate if that's the case because then I wasted a bunch of time, mine and yours. That's all I have to say. Thank you for your time

**Comment by wowteam1** [comment]
How much proof can you need? Its right there. Fidbugs is right.

Both gauntlets blazefury and fiery retributer are harmful damage spells.

The proof is

1. Storm gauntlets EATS your stormstrike with its nature damage spell

2. Grounding totem that is destroyed by harmful spells is destroyed by storm gauntlets nature damage

They are harmful spells and therefore affect wrath of cenarius

**Comment by Gagarinbest** [comment]
How noob made an impact on Blizzard to upgrade some ring in WoW Pre-Patch TBC Classic

Dear comrades! 

I want to share with you my observation about min-maxing Wrath of Cenarius ring proc, which was significantly improved just now, especially for TBC Pre-Patch. Now, it could have an amazing uptime, no internal cooldown and proc (+132 SP for 10 sec) even from Cloak of Fire, ordinary Seal of Righteousness, Crusader Strike, or Frost/Shadow oils.

Also, it could proc from Fiery Retributer, Fiery Plate / Storm Gauntlets and possible Blazefurry Medallion attacks. All of these items could crit, i.e., it's a valuable source for your Vengeance talent as well.

On mid December 2025 , I tested it, and it didn’t proc from the above-mentioned items and abilities. After this, I wrote a letter to Blizzard with some recommendations and miracles that could happened… Now, you can see it all in the video below. 

Btw, the quest mobs (insects/bugs) in Silithus for this Wrath of Cenarius ring during the Pre-Patch should become non-elite, so now it’s not so hard to obtain, as earlier. I want to say thanks to Jesus Christ and blizzard's developers for this upgrade.

For all paladins, especially Prot and Ret, and shamans worldwide.

Enjoy!

https://youtu.be/tuIICb6tUR4?si=uSDyQurZ64OzZzC1
